src/material-experimental/mdc-form-field/_form-field-density.scss

```diff
@@ -0,0 +1,115 @@
+@import '@material/density/functions.import';
+@import '@material/textfield/variables.import';
+@import 'form-field-sizing';
+
+// Mixin that sets the vertical spacing for the infix container of filled form fields.
+// We need to apply spacing to the infix container because we removed the input padding
+// provided by MDC in order to support arbitrary form-field controls.
+@mixin _mat-form-field-infix-vertical-spacing-filled($with-label-padding, $no-label-padding) {
+  .mat-mdc-text-field-wrapper:not(.mdc-text-field--outlined) .mat-mdc-form-field-infix {
```

Comment:
is there no way to do this with less specific selectors?

Reply:
It might be possible to reduce specificity if we add a class binding for filled form-fields. Not sure if that is worth though since MDC textfield uses the same pattern and generally has high specificity selectors.

Reply:
Just played with it. The specificity could be reduced by a bit here by adding a new class binding, but I'm unsure whether it's worth given that all other part of the form-field/text-field have high specificity due to MDC. Happy to change if you feel strongly about.

FWIW: This logic just moved from another file, so it's not necessarily belonging into the `density-api` branch I guess.

Reply:
ok, I think it's fine if MDC already has this kind of thing